pkg/provider/cloud/aws/policy.go

```diff
@@ -49,149 +53,304 @@ var (
       "Resource": [
         "*"
       ]
-    }
-  ]
-}
-`
+    },
 
-	// The role for control plane.
-	// Based on https://github.com/kubernetes/kops/blob/master/docs/iam_roles.md
-	// We're using 2 filters:
-	// - RequestTag  = Makes sure the tag exists on the create request
-	// - ResourceTag = Makes sure the tag exists on the resource that should be modified
-	// Actions are grouped by ability of filtering:
-	// - Actions which cannot be filtered (Setting a filter will make them fail)
-	// - EC2 create actions with the RequestTag filter
-	// - EC2 modify actions with the ResourceTag filter
-	// - ELB create actions with the RequestTag filter
-	// - ELB modify actions with the ResourceTag filter.
-	controlPlanePolicyTpl = template.Must(template.New("worker-policy").Parse(`{
-  "Version": "2012-10-17",
-  "Statement": [
     {
       "Effect": "Allow",
       "Action": [
+        "ec2:CreateSnapshot",
+        "ec2:AttachVolume",
+        "ec2:DetachVolume",
+        "ec2:ModifyVolume",
+        "ec2:DescribeAvailabilityZones",
         "ec2:DescribeInstances",
-        "ec2:DescribeRegions",
-        "ec2:DescribeRouteTables",
-        "ec2:DescribeSecurityGroups",
-        "ec2:DescribeSubnets",
+        "ec2:DescribeSnapshots",
+        "ec2:DescribeTags",
         "ec2:DescribeVolumes",
-        "ec2:CreateSecurityGroup",
-        "ec2:DescribeVolumesModifications",
-        "ec2:ModifyInstanceAttribute",
-        "ec2:ModifyVolume",
-        "ec2:DescribeVpcs"
+        "ec2:DescribeVolumesModifications"
       ],
-      "Resource": [
-        "*"
-      ]
+      "Resource": "*"
     },
     {
       "Effect": "Allow",
       "Action": [
-        "ec2:CreateVolume",
-        "ec2:CreateRoute"
+        "ec2:CreateTags"
       ],
       "Resource": [
-        "*"
+        "arn:aws:ec2:*:*:volume/*",
+        "arn:aws:ec2:*:*:snapshot/*"
       ],
       "Condition": {
-        "Null": {
-          "ec2:RequestTag/{{ .ClusterTag }}": "false"
+        "StringEquals": {
+          "ec2:CreateAction": [
+            "CreateVolume",
+            "CreateSnapshot"
+          ]
         }
       }
     },
     {
       "Effect": "Allow",
       "Action": [
-        "ec2:CreateTags",
-        "ec2:AttachVolume",
-        "ec2:AuthorizeSecurityGroupIngress",
-        "ec2:DeleteRoute",
-        "ec2:DeleteSecurityGroup",
-        "ec2:DeleteVolume",
-        "ec2:DetachVolume",
-        "ec2:RevokeSecurityGroupIngress"
+        "ec2:DeleteTags"
       ],
       "Resource": [
-        "*"
+        "arn:aws:ec2:*:*:volume/*",
+        "arn:aws:ec2:*:*:snapshot/*"
+      ]
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "ec2:CreateVolume"
       ],
+      "Resource": "*",
       "Condition": {
-        "Null": {
-          "ec2:ResourceTag/{{ .ClusterTag }}": "false"
+        "StringLike": {
+          "aws:RequestTag/{{ .EBSCSIClusterTag }}": "true"
```

Comment:
How is this going to behave when we use the in-tree cloud provider? In that case, is `ebs.csi.aws.com/$CLUSTER_NAME` a valid condition since `kubernetes.io/aws-ebs` is used as a provisioner instead?

Reply:
![screenshot-2022-10-05-132212](https://user-images.githubusercontent.com/127499/194049207-fb52a796-0320-4724-b0c4-e9c587060e2e.png)

The new policy contains variants for both the old and the new tags, so I think we should be fine. This comes straight from upstream: https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/docs/example-iam-policy.json#L48-L83

Reply:
@xrstf Thanks for the clarification!